packages/react/src/elements/components/Banner/BannerContainer.js

```diff
@@ -0,0 +1,215 @@
+    isWrappingActions: false,
+    isWrappingContent: false
+  };
+
+  componentDidMount() {
+    this.bindResize();
+    this.updateWrapping();
+  }
+
+  componentWillUnmount() {
+    this.unbindResize();
+    this.cancelWrappingUpdate();
+  }
+
+  /** @type {BannerContainerProps | any} */
+  props;
+  /** @type {HTMLDivElement | undefined} */
+  content;
+  /** @type {HTMLDivElement | undefined} */
+  interactionsWrapper;
+  /** @type {HTMLParagraphElement | undefined} */
+  notification;
+  /** @type {number | undefined} */
+  wrappingFrame;
+
+  /**
+   * @param {HTMLDivElement} element
+   */
+  refContent = element => {
+    this.content = element;
+  };
+
+  /**
+   * @param {HTMLParagraphElement} element
+   */
+  refNotification = element => {
+    this.notification = element;
+  };
+
+  /**
+   * @param {HTMLDivElement} element
+   */
+  refInteractionsWrapper = element => {
+    this.interactionsWrapper = element;
+  };
+
+  handleResize = () => {
+    this.updateWrapping();
+  };
+
+  /**
+   * Determines whether actions should wrap on the next repaint
+   * @returns {boolean}
+   */
+  shouldWrapActions() {
+    const { interactionsWrapper } = this;
+
+    if (!interactionsWrapper) return false;
+
+    const { isWrappingActions } = this.state;
+    const notificationWidth = isWrappingActions
+      ? ACTIONS_WRAPPING_THRESHOLD + interactionsWrapper.offsetWidth * 2
+      : ACTIONS_WRAPPING_THRESHOLD;
+
+    return this.willContentOverflow(notificationWidth);
+  }
+
+  /**
+   * Determines whether content should wrap on the next repaint
+   * @returns {boolean}
+   */
+  shouldWrapContent() {
+    return this.willContentOverflow(MIN_NOTIFICATION_WIDTH);
+  }
+
+  /**
+   * Determines whether the content will overflow its container
+   * with the given width of the notification element.
+   *
+   * @param {number} notificationWidth
+   * @returns {boolean}
+   */
+  willContentOverflow(notificationWidth) {
+    const { content, interactionsWrapper } = this;
+
+    if (!content || !interactionsWrapper) return false;
+
+    const contentWidth = content.offsetWidth;
+    const actionsWidth = interactionsWrapper.offsetWidth;
+    const contentChildrenWidth =
+      notificationWidth + MIN_CONTENT_SPACING + actionsWidth;
+
+    return contentChildrenWidth > contentWidth;
+  }
+
+  updateContentWrapping = () => {
+    const update = { isWrappingContent: this.shouldWrapContent() };
+
+    this.setState(update, () => {
+      delete this.wrappingFrame;
+    });
+  };
+
+  updateActionWrapping = () => {
+    const update = { isWrappingActions: this.shouldWrapActions() };
+
+    this.setState(update, () => {
+      this.wrappingFrame = window.requestAnimationFrame(
+        this.updateContentWrapping
+      );
+    });
+  };
+
+  /**
+   * Asynchronously updates the wrapping behavior of the presenter
+   */
+  updateWrapping() {
+    if (this.wrappingFrame !== undefined) return;
+
+    this.wrappingFrame = window.requestAnimationFrame(
```

Comment:
As discussed, this throttles the number of times the browser tries to repaint the Banner when resizing or scrolling.